cherry-pick-2.0: ui: A collection of fixes for the network debugging page

BramGruneir opened the pull request:
cherry-pick of #23844

---

- Fix the case in which longs had no value during calculation causing the page
to not load.
- Fix latency page to only display when there is at least 2 nodes worth of
latencies to display.
- Add a loading spinner to the network diagnostics page.
- Fix the loading spinner to accept an array of children instead of just a
single child element.
- Add a new top padded style so the loading spinner is not so cramped on the
page.
- Fix a crash when a new node doesn't yet have any activity.
- Replace all not yet calculated latencies with a `?`.
- Don't display a legend if there is only one value (and hence no stddev).

Fixes #23552.

Release note (admin ui change): The network diagnositcs report will no longer
crash when the latencies are very small or on a single node cluster.

BramGruneir committed:
ui: A collection of fixes for the network debugging page

- Fix the case in which longs had no value during calculation causing the page
to not load.
- Fix latency page to only display when there is at least 2 nodes worth of
latencies to display.
- Add a loading spinner to the network diagnostics page.
- Fix the loading spinner to accept an array of children instead of just a
single child element.
- Add a new top padded style so the loading spinner is not so cramped on the
page.
- Fix a crash when a new node doesn't yet have any activity.
- Replace all not yet calculated latencies with a `?`.
- Don't display a legend if there is only one value (and hence no stddev).

Fixes #23552.

Release note (admin ui change): The network diagnositcs report will no longer
crash when the latencies are very small or on a single node cluster.

vilterp commented:
Somehow this broke the node map. On master, node map works before this commit and shows up blank after it. Let's hold off on cherry picking until we fix master.

Commit:
ui: unbreak cluster viz by removing extraneous div in Loading

Unfortunately, the cluster viz styles are brittle enough that the change
in #23844 which introduced an additional <div> as a child of the
<Loading> component broke them, causing the node map to have zero width
and height.

Would be nice to make those less brittle, but in the short term this
commit enforces that only one child is passed to the Loading component,
and moves the container div to the Network page (which motivated the
multi-child case).

Release note: None

vilterp commented:
K, have a fix: #23879

BramGruneir commented:
This now includes #23879 so that cluster viz is no longer broken.
